### PR TITLE
new crowdsec w/ freebsd 13

### PR DESCRIPTION
**Pull request by mmetc**
from https://github.com/sbz/packaging-freebsd/issues/1

- Bump distribution target: FreeBSD 12.2 -> 13.0

- Bump Crowdsec release: 1.1.1 -> 1.2.0 and firewall-bouncer 0.0.13 -> 0.0.16

- Fixed path issues. Go code in CrowdSec has hardcoded paths for both /etc/crowdsec and /var/lib. We will eventually fix the issue at the source level but in the meantime, sed replacements will do. I extended them to more source files and included the data directory (/var/lib -> /var/db)

- More configuration files provided as ".sample": acquis.yaml, profiles.yaml, simulation.yaml, local_api_credentials and online_api_credentials. The last three are usually not edited by users but need nevertheless to be maintained across updates.

- The port's Makefile builds the CLI binary as crowdsec-cli, but all the Linux ports and the CrowdSec documentation refer to it as cscli, so I renamed the binary after the build.

- The RC script now connects (upon first use) to fetch the hub's inventory, register local/online APIs and download GeoIP data, all while querying cscli to read the right pathnames from the configuration.

- Updated post-install message to tell the user to also edit acquis.yaml and profiles.yaml in addition to config.yaml because we don't detect the running services in FreeBSD (no wizard.sh)

- Deinstall script to stop the service before removing the package.

### CODE REVIEW

File: security/crowdsec/files/crowdsec.in [hunk 1]
```diff
@@ -35,14 +35,32 @@ configtest_cmd="${name}_configtest"
 extra_commands="configtest reload"
 
 crowdsec_precmd() {
-    if [ ! -d %%PREFIX%%/etc/crowdsec/hub ]; then
-        %%PREFIX%%/bin/crowdsec-cli hub update || :
+    HUB_DIR=`%%PREFIX%%/bin/cscli config show | grep 'Hub Folder' | head -1 | awk -F: '{gsub(/: /,":"); printf "%s", $2}'`
```
Comment (sbz) [inline security/crowdsec/files/crowdsec.in:38]: Can you just limit number of pipes here using just `awk` or `sed`?  or maybe you can rely on the output `-o raw` to ease the parsing of the config value.

File: security/crowdsec/files/pkg-deinstall.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+case $2 in
+       DEINSTALL)
+               %%PREFIX%%/etc/rc.d/crowdsec stop
```
Comment (sbz) [inline security/crowdsec/files/pkg-deinstall.in]: We could either use the command `service` or the rc script here both are valid options.

File: security/crowdsec/files/crowdsec.in (hunk 1 above)
Comment (mmetc) [reply security/crowdsec/files/crowdsec.in:38]: I know, I couldn't do it simpler and the configuration option could appear twice in the output. Upon my request, the next version of CrowdSec has a "config show <specific_option>" to display only what I want, so all of those pipes will go away.

File: security/crowdsec/files/pkg-deinstall.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+               ;;
+       POST-DEINSTALL)
```
Comment (sbz) [inline security/crowdsec/files/pkg-deinstall.in]: The `POST-DEINSTALL` case could be used to clean extraneous leftovers such as crowdsec agent pid file, data dir, etc.. Basically everything created but not handled by the package files listing.

If it's not used at the moment, I would say not need to add it.

**Review by sbz — APPROVED** [review]
I added some comments but nothing prevent to merge it at this point.

Great job, thanks! 👍

File: security/crowdsec/files/pkg-deinstall.in (hunk 2 above)
Comment (mmetc) [reply security/crowdsec/files/pkg-deinstall.in]: "service crowdsec stop", ok. I was doing like other projects in /src/ports

**Pull request merged**